docs/howtos/extending/index.md

```diff
@@ -0,0 +1,20 @@
+(extending-napari)=
+# Extending the `napari` GUI
+
+There are a number of ways to extend your usage of `napari`:
+
+* [**Adding widgets**](creating-widgets): adding a widget (new GUI interaction element)
+  to your `napari` app.
+* **Connecting callbacks**: customize the behavior of `napari` by connecting an action
+  (function) to an [event](intro-to-event-loop) such as:
+  - [**Key press**](connect-key-event): perform an action based on a keyboard shortcut
+  - [**Mouse event**](connect-mouse-event): perform an action based on a mouse event
+  - [**Layer/viewer event**](connect-napari-event): perform an action when a layer or
```

Comment:
I kinda want to split responding to `Layer/viewer events` with custom callbacks and adding your own key press/mouse event handlers, but maybe there's not enough distinction? With key preses/mouse events you're defining both the event and the callback, so you could be introducing new events even. While responding to layer/viewer events is exactly that: you're hooking into existing events emitted by the viewer to perform custom actions. But maybe it's not really worth distinguishing idk

Reply:
I see that those are all in the same doc anyway so maybe just ignore this